content/commands/hexpire/index.md

```diff
@@ -0,0 +1,139 @@
+---
+acl_categories:
+- '@write'
+- '@hash'
+- '@fast'
+arguments:
+- display_text: key
+  key_spec_index: 0
+  name: key
+  type: key
+- display_text: seconds
+  name: seconds
+  type: integer
+- arguments:
+  - display_text: nx
+    name: nx
+    token: NX
+    type: pure-token
+  - display_text: xx
+    name: xx
+    token: XX
+    type: pure-token
+  - display_text: gt
+    name: gt
+    token: GT
+    type: pure-token
+  - display_text: lt
+    name: lt
+    token: LT
+    type: pure-token
+  name: condition
+  optional: true
+  type: oneof
+- display_text: numfields
+  name: numfields
+  type: integer
+- display_text: field
+  multiple: true
+  name: field
+  type: string
+arity: -5
+categories:
+- docs
+- develop
+- stack
+- oss
+- rs
+- rc
+- oss
+- kubernetes
+- clients
+command_flags:
+- write
+- denyoom
+- fast
+complexity: O(N) where N is the number of arguments to the command
+description: Set expiration for hash fields using relative time to expire (seconds)
+group: hash
+hidden: false
+key_specs:
+- RW: true
+  begin_search:
+    spec:
+      index: 1
+    type: index
+  find_keys:
+    spec:
+      keystep: 1
+      lastkey: 0
+      limit: 0
+    type: range
+  update: true
+linkTitle: HEXPIRE
+since: 8.0.0
+summary: Set expiration for hash fields using relative time to expire (seconds)
+syntax_fmt: HEXPIRE key seconds [NX | XX | GT | LT] numfields field [field ...]
+syntax_str: seconds [NX | XX | GT | LT] numfields field [field ...]
+title: HEXPIRE
+---
+Set an expiration (TTL or time to live) on one or more fields of a given hash key. At least one field must be specified.
+After the TTL has expired, the field(s) will automatically be deleted from the hash key.
+
+Field expirations will only be cleared by commands that delete or overwrite the
+contents of the hash key fields, including [`HDEL`]({{< relref "/commands/hdel" >}}) and [`HSET`]({{< relref "/commands/hset" >}})
+commands.
+This means that all the operations that conceptually _alter_ the value stored at a hash key's field without replacing it with a new one will leave the TTL untouched.
+
+The TTL can be cleared using the [`HPERSIST`]({{< relref "/commands/hpersist" >}}) command, turning the hash key field back into a persistent field.
+
+Note that calling `HEXPIRE`/[`HPEXPIRE`]({{< relref "/commands/hpexpire" >}}) with a non-positive TTL or
+[`HEXPIREAT`]({{< relref "/commands/hexpireat" >}})/[`HPEXPIREAT`]({{< relref "/commands/hpexpireat" >}}) with a time in the past will result in the key field being
+[deleted]({{< relref "/commands/hdel" >}}) rather than expired (accordingly, the emitted [key event]({{< relref "/develop/use/keyspace-notifications" >}})
```

Comment:
"key being deleted rather than expired" is misleading. The field will still expire - just sooner (there are keyspace notifications in Redis - and you would still get an expiration notification and not a deletion notification).

Reply:
We'll need to update https://redis.io/docs/latest/develop/use/keyspace-notifications/ with field notifications. I don't have the details. R&D worked on it.

Reply:
Note that when the last field is expired/deleted - the key is deleted because a hash can't be empty.

Reply:
https://github.com/redis/redis/pull/13237

Reply:
Keyspace notification page has been updated [here](https://github.com/redis/docs/pull/238) .